ironic/drivers/modules/drac/raid.py

```diff
@@ -996,6 +1064,44 @@ def _create_virtual_disks(task, node):
     return _commit_to_controllers(node, controllers)
 
 
+def controller_supports_ehba_mode(settings, controller_fqdd):
+    # check the raid controller mode
+    raid_controller_mode = settings.get(
+        "{}:RAIDCurrentControllerMode".format(controller_fqdd))
+    raid_controller_mode = raid_controller_mode.current_value
+
+    # if e-HBA found then return True
+    if "Enhanced HBA" in raid_controller_mode:
+        return True
+    else:
+        return False
+
+
+def _config_raid_settings(node):
+    raid_settings = list_raid_settings(node)
+    controllers = list_raid_controllers(node)
+    raid_controllers = [cntrl for cntrl in controllers
+                        if cntrl.model.startswith('PERC H740P')
+                        and controller_supports_ehba_mode(raid_settings,
+                                                          cntrl.id)]
+
+    controllers = list()
+    for controller in raid_controllers:
+        raid_attr = "{}:RAIDRequestedControllerMode".format(controller.id)
```

Comment:
Will need to check here if the controller is in HBA mode and set it to RAID mode if not.

Reply:
I have one question for this: we are iterating this `raid_controllers` which are defined at line 1083 on above comment. In that we are already checking whether raid controller support HBA mode or not. Do we need to check it again at this line?

Reply:
Well, here, we need to check if the RAID controller is currently in HBA mode.  If it is, then we need to take it out of HBA mode, otherwise, we don't need to do anything to it.  In the method above, we'll check if the controller supports HBA mode, meaning, is it capable of being put into HBA mode.

Reply:
Got your point. Thanks for clarification.

Reply:
Done